### PR TITLE
add get_net_size

### PR DESCRIPTION
**Pull request by g1y5x3**
Let me know if this is what you are looking for in https://github.com/tinygrad/tinygrad/pull/2041#issuecomment-1760035006
```
(tiny_env) [g1y5x3@tiny3:tinygrad]$ HALF=0 GPU=1 STEPS=1 python examples/hlb_cifar10.py
Layers                Params     Bytes
------------------  --------  --------
whitening                144       576
net.0.weight             384      1536
net.2.conv1.weight     18432     73728
net.2.conv2.weight     36864    147456
net.2.norm1.weight        64       256
net.2.norm1.bias          64       256
net.2.norm2.weight        64       256
net.2.norm2.bias          64       256
net.3.conv1.weight    147456    589824
net.3.conv2.weight    589824   2359296
net.3.norm1.weight       256      1024
net.3.norm1.bias         256      1024
net.3.norm2.weight       256      1024
net.3.norm2.bias         256      1024
net.4.conv1.weight   1179648   4718592
net.4.conv2.weight   2359296   9437184
net.4.norm1.weight       512      2048
net.4.norm1.bias         512      2048
net.4.norm2.weight       512      2048
net.4.norm2.bias         512      2048
net.6.weight            5120     20480
Total                4340496  17361984
```
```
(tiny_env) [g1y5x3@tiny3:tinygrad]$ HALF=1 GPU=1 STEPS=1 python examples/hlb_cifar10.py
Layers                Params    Bytes
------------------  --------  -------
whitening                144      288
net.0.weight             384      768
net.2.conv1.weight     18432    36864
net.2.conv2.weight     36864    73728
net.2.norm1.weight        64      128
net.2.norm1.bias          64      128
net.2.norm2.weight        64      128
net.2.norm2.bias          64      128
net.3.conv1.weight    147456   294912
net.3.conv2.weight    589824  1179648
net.3.norm1.weight       256      512
net.3.norm1.bias         256      512
net.3.norm2.weight       256      512
net.3.norm2.bias         256      512
net.4.conv1.weight   1179648  2359296
net.4.conv2.weight   2359296  4718592
net.4.norm1.weight       512     1024
net.4.norm1.bias         512     1024
net.4.norm2.weight       512     1024
net.4.norm2.bias         512     1024
net.6.weight            5120    10240
Total                4340496  8680992
```

### CODE REVIEW

**Comment by geohot** [comment]
Oh this is more than I was thinking, I was thinking to output like one line.

File: extra/helpers.py [hunk 1]
```diff
@@ -1,6 +1,15 @@
 import multiprocessing, subprocess
 import cloudpickle  # type: ignore
 from typing import Any
+from tinygrad.nn.state import get_state_dict
+
+def get_net_size(model):
+  total_params, total_bytes = 0, 0
+  for param_name, param in zip(get_state_dict(model), get_state_dict(model).values()):
+    if not ("num_batches_tracked" in param_name) and not ("running" in param_name):
```
Comment (geohot) [inline extra/helpers.py:9]: Err, I don't like this gate based on name.

Also, huh?!?
`zip(get_state_dict(model), get_state_dict(model).values()):`

Isn't that just:
`get_state_dict(model).items()`

Comment (g1y5x3) [reply extra/helpers.py:9]: you are right. thank you for catching this.

File: extra/helpers.py [hunk 2]
```diff
@@ -1,6 +1,15 @@
 import multiprocessing, subprocess
 import cloudpickle  # type: ignore
 from typing import Any
+from tinygrad.nn.state import get_state_dict
+
+def get_net_size(model):
+  total_params, total_bytes = 0, 0
+  for param_name, param in get_state_dict(model).items():
+    if not ("num_batches_tracked" in param_name) and not ("running" in param_name):
```
Comment (geohot) [inline extra/helpers.py:9]: This still isn't generic though, I don't like the name gate. If it can't be generic maybe we don't want it.

**Comment by geohot** [comment]
Going to close, I'm not sure about the generic usefulness of this

**Pull request closed without merging**